unittests/Parser/ParserTests.cpp

```diff
@@ -70,6 +70,22 @@ TEST(ParserTest, Errors) {
         "<input>:2:1: cttz doesn't support 70 bits" },
       { "%0:i128 = var ; 0\n%1:i128 = ctlz %0\n",
         "<input>:2:1: ctlz doesn't support 128 bits" },
+      { "%0:i4 = var (00010)\n",
+        "<input>:1:1: knownbits pattern must be of same length as var width" },
+      { "%0:i4 = var (1xx00)\n",
+        "<input>:1:1: knownbits pattern must be of same length as var width" },
+      { "%0:i4 = var (xx1)\n",
+        "<input>:1:1: knownbits pattern must be of same length as var width" },
+      { "%0:i4 = var (0012)\n",
+        "<input>:1:17: invalid knownbits string" },
+      { "%0:i4 = var (2345)\n",
+        "<input>:1:14: invalid knownbits string" },
+      { "%0:i4 = var (01xa)\n",
+        "<input>:1:17: invalid knownbits string" },
+      { "%0:i4 = var (xxx)\n",
+        "<input>:1:1: knownbits pattern must be of same length as var width" },
+      { "%0:i4 = var ()\n",
+        "<input>:1:14: invalid knownbits string" },
 
```

Comment:
you should add a test case where the line ends and there's no closing parenthesis